build.zig

```diff
@@ -74,26 +70,44 @@ pub fn build(b: *std.Build) void {
 
     if (mbedtls) {
         ssh2_lib.root_module.addCMacro("LIBSSH2_MBEDTLS", "1");
-        ssh2_lib.linkSystemLibrary2("mbedtls", .{ .preferred_link_mode = mbedtls_linkage });
-        ssh2_lib.linkSystemLibrary2("mbedcrypto", .{ .preferred_link_mode = mbedtls_linkage });
-        ssh2_lib.linkSystemLibrary2("mbedx509", .{ .preferred_link_mode = mbedtls_linkage });
+        if (b.systemIntegrationOption("mbedtls", .{ .default = true })) {
+            ssh2_lib.linkSystemLibrary("mbedtls");
+            ssh2_lib.linkSystemLibrary("mbedcrypto");
+            ssh2_lib.linkSystemLibrary("mbedx509");
+        } else {
+            // TODO: Add lazy dependency to build.zig.zon for linking statically.
+            // For now it's the users resposibility to compile and statically link against library
```

Comment:
As mentioned, this has nothing to do with static/dynamic linking just means we don't try to link the system library for folks automatically

We can provide an error message here maybe:

```suggestion
            @panic("mbedtls not provided. If you pass --no-sys=mbedtls you must link your own mbedtls to the final app.");
```

Reply:
I'm not sure if a nice error message can be provided, because using `@panic` here will make it always show the message. 

Currently the error you will get when it isn't linked is this:
```shell
/home/rokas/.cache/zig/p/N-V-__8AAATRLQBu_rNy4X2UK6RtcdYK_yAzkV6OcCqgo5aK/src/mbedtls.h:59:10: error: 'mbedtls/version.h' file not found
#include <mbedtls/version.h>
         ^~~~~~~~~~~~~~~~~~~~
/home/rokas/.cache/zig/p/N-V-__8AAATRLQBu_rNy4X2UK6RtcdYK_yAzkV6OcCqgo5aK/src/agent.c:42:10: note: in file included from /home/rokas/.cache/zig/p/N-V-__8AAATRLQBu_rNy4X2UK6RtcdYK_yAzkV6OcCqgo5aK/src/agent.c:42:
#include "libssh2_priv.h"
         ^
/home/rokas/.cache/zig/p/N-V-__8AAATRLQBu_rNy4X2UK6RtcdYK_yAzkV6OcCqgo5aK/src/libssh2_priv.h:187:10: note: in file included from /home/rokas/.cache/zig/p/N-V-__8AAATRLQBu_rNy4X2UK6RtcdYK_yAzkV6OcCqgo5aK/src/libssh2_priv.h:187:
#include "crypto.h"
         ^
/home/rokas/.cache/zig/p/N-V-__8AAATRLQBu_rNy4X2UK6RtcdYK_yAzkV6OcCqgo5aK/src/crypto.h:49:10: note: in file included from /home/rokas/.cache/zig/p/N-V-__8AAATRLQBu_rNy4X2UK6RtcdYK_yAzkV6OcCqgo5aK/src/crypto.h:49:
#include "mbedtls.h"

// ...
```